examples/ensemble/plot_forest_hist_grad_boosting_comparison.py

```diff
@@ -0,0 +1,210 @@
+===============================================================
+
+In this example we compare the performance of Random Forest (RF) and Histogram
+Gradient Boosting (HGBT) models in terms of score and computation time for a
+given dataset. Here we vary the parameters that control the number of trees
+according to each estimator:
+
+- `n_estimators` controls the number of trees in the forest. It's a fixed numer.
+- `max_iter` is the the maximum number of iterations in a gradient boosting
+  based model. The number of iterations corresponds to the number of trees for
+  regression and binary classification problems. Furthermore, the actual number
+  of trees required by the model depends on the stopping criteria.
+
+HGBT use gradient boosting to iteratively improve the model's performance by
+fitting each tree to the negative gradient of the loss function with respect to
+the predicted value. RFs, on the other hand, are based on bagging and use a
+majority vote to predict the outcome.
+
+For more information on ensemble models, see the :ref:`User Guide <ensemble>`.
+"""
+
+# Author:  Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+# License: BSD 3 clause
+
+# %%
+# Load dataset
+# ------------
+
+from sklearn.datasets import fetch_california_housing
+
+X, y = fetch_california_housing(return_X_y=True, as_frame=True)
+n_samples, n_features = X.shape
+
+# %%
+# HGBT uses a binning algorithm that can efficiently handle large datasets (tens
+# of thousands samples or more) with a high number of features (see
+# :ref:`Why_it's_faster`). The scikit-learn implementation of RF does not use
+# binning and relies on exact splitting, which can be computationally expensive.
+
+print(f"The dataset consists of {n_samples} samples and {n_features} features")
+
+# %%
+# Compute score and computation times
+# -----------------------------------
+#
+# Notice that many parts of the implementation of
+# :class:`~sklearn.ensemble.HistGradientBoostingClassifier` and
+# :class:`~sklearn.ensemble.HistGradientBoostingRegressor` are parallelized by
+# default.
+#
+# The implementation of :class:`~sklearn.ensemble.RandomForestRegressor` and
+# :class:`~sklearn.ensemble.RandomForestClassifier` can also be run on multiple
+# cores by using the `n_jobs` parameter, here set to 2 due to limitations on the
+# documentation builder. See :ref:`parallelism` for more information.
+
+import joblib
+
+N_CORES = joblib.cpu_count(only_physical_cores=True)
+print(f"Number of physical cores: {N_CORES}")
+
+# %%
+import threadpoolctl
+
+threadpoolctl.threadpool_info()
+
+# %%
+# The other parameters of both models were tuned but the procedure is not shown
+# here to keep the example simple.
+
+import pandas as pd
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.model_selection import GridSearchCV, KFold
+
+models = {
+    "Random Forest": RandomForestRegressor(
+        min_samples_leaf=5, random_state=0, n_jobs=2
```

Comment:
To be fair to RF we should actually use:

```suggestion
        min_samples_leaf=5, random_state=0, n_jobs=N_CORES
```

Reply:
This should not impact the performance on the CI where `N_CORES` is 2 but it will give a fairer picture on local machines.